homeassistant/components/openweathermap/weather.py

```diff
@@ -49,7 +53,11 @@ class OpenWeatherMapWeather(WeatherEntity):
 
     _attr_attribution = ATTRIBUTION
     _attr_should_poll = False
-    _attr_temperature_unit = TEMP_CELSIUS
+
+    _attr_native_precipitation_unit = LENGTH_MILLIMETERS
```

Comment:
owm uses mm for precipitation: https://pyowm.readthedocs.io/en/latest/v3/code-recipes.html